src/qibolab/backends.py

```diff
@@ -50,6 +51,25 @@ def __init__(self, platform):
         }
         self.compiler = Compiler.default()
 
+    @property
+    def qubits(self) -> list[Union[str, int]]:
+        return list(self.platform.qubits.keys())
+
+    @property
+    def connectivity(self) -> list[tuple[Union[str, int], Union[str, int]]]:
+        return list(self.platform.pairs)
+
+    @property
+    def natives(self) -> list[str]:    
+        native_gates = set()
+        for _, q in self.platform.qubits.items():
+            native_gates |= {k for k, v in q.native_gates.__dict__.items() if v is not None}
```

Comment:
In general, try to avoid as much as possible to use dunders (i.e. methods like `__method__`) since they may be internal implementations of some protocol, but possibly prone to changes (ok, `__dict__` won't change soon, but it may be used differently by certain classes).

In this case
https://github.com/qiboteam/qibolab/blob/30db7ea9f63949c5390effa41a240d0626c9b3ae/src/qibolab/native.py#L258-L259
and since it is a dataclass, you can iterate its fields with the [`fields`](https://docs.python.org/3/library/dataclasses.html#dataclasses.fields) function, as done in
https://github.com/qiboteam/qibolab/blob/30db7ea9f63949c5390effa41a240d0626c9b3ae/src/qibolab/native.py#L298-L299

Otherwise, it may be simpler in this case to just use the [`asdict()`](https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict) function, which will just replace your usage of `__dict__`
```suggestion
            native_gates |= {k for k, v in asdict(q.native_gates).items() if v is not None}
```